python/llama-index-server/llama_index/server/api/routers/chat.py

```diff
@@ -129,6 +179,25 @@ async def _text_stream(
             elif hasattr(event, "to_response"):
                 event_response = event.to_response()
                 yield VercelStreamResponse.convert_data(event_response)
+            elif isinstance(event, InputRequiredEvent):
+                run_id = handler.workflow_handler.run_id
+                if run_id is None:
+                    raise RuntimeError("Run ID is None")
+                ctx = handler.workflow_handler.ctx
+                if ctx is None:
+                    raise RuntimeError("Context is None")
+                run_id = WorkflowService.save_context(
+                    run_id=run_id,
+                    ctx=ctx,
```

Comment:
how about save_context generates ID?
```suggestion
                run_id = WorkflowService.save_context(
                    ctx=ctx,
```